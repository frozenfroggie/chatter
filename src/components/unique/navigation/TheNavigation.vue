<template>
  <div class='navigation'>
    <the-navigation-avatar/>
    <nav-btn-conversations/>
    <nav-btn-friends/>
    <nav-btn-add-friends/>
    <nav-btn-settings/>
    <nav-btn-logout v-on:logout='logout'/>
  </div>
</template>

<script>
import TheNavigationAvatar from './TheNavigationAvatar'
import NavBtnConversations from './navigationButtons/NavBtnConversations'
import NavBtnFriends from './navigationButtons/NavBtnFriends'
import NavBtnAddFriends from './navigationButtons/NavBtnAddFriends'
import NavBtnSettings from './navigationButtons/NavBtnSettings'
import NavBtnLogout from './navigationButtons/NavBtnLogout'

export default {
  components: {
    TheNavigationAvatar,
    NavBtnConversations,
    NavBtnFriends,
    NavBtnAddFriends,
    NavBtnSettings,
    NavBtnLogout
  },
  methods: {
    logout () {
      // window.localStorage.clear()
      this.$store.dispatch('logout')
      this.$router.push('/')
    }
  }
}
</script>

<style scoped>
  .navigation {
    position: fixed;
    top: 0px;
    left: 0px;
    bottom: 0px;
    width: 46px;
    display: grid;
    grid-template-columns: auto;
    grid-template-rows: 2fr 1fr 1fr 1fr 1fr 3fr;
    grid-column-gap: 10px;
    grid-row-gap: 10px;
    grid-template-areas:
      "avatar"
      "conversations"
      "friends"
      "addFriends"
      "settings"
      "logout";
    background-color: #2A384A;
    z-index: 15;
    box-shadow: 0px 6px 20px 0px rgba(0,0,0,0.7);
  }
</style>
