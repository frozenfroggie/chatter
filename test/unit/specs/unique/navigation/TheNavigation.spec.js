import { shallowMount, mount, RouterLinkStub } from '@vue/test-utils'
import TheNavigation from '@/components/unique/navigation/TheNavigation'

describe('TheNavigation', () => {
  it('matches snapshot', () => {
    const wrapper = mount(TheNavigation, {
      stubs: {
        icon: true,
        RouterLink: RouterLinkStub
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
