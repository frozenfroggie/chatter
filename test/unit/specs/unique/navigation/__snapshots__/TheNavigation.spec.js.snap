// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheNavigation matches snapshot 1`] = `
<div class="navigation">
    <div class="avatarContainer">
        <!---->
        <div class="online"></div>
    </div>
    <div class="conversations">
        <!---->
    </div>
    <div class="friends">
        <!---->
    </div>
    <div class="addFriends">
        <!---->
    </div>
    <div class="settings">
        <!---->
    </div>
    <div class="logout">
        <!---->
    </div>
</div>
`;
